bindings/pprof.cc

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright 2021 Datadog Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pprof.h"
+
+#include <iostream>
+
+namespace pprof {
+
+//
+// Encoding
+//
+Encoder::Encoder() {
+  string_table.push_back("");
+}
+
+uint64_t Encoder::dedup(const std::string& str) {
+  auto found = std::find(string_table.begin(), string_table.end(), str);
```

Comment:
Would we not prefer a flat set here ?

Reply:
It needs to return an index where the string is stored so the caller of `dedup(...)` can reference it though. As far as I know, sets don't follow insertion order.